pkg/cover/store.go

```
@@ -153,8 +167,32 @@ func (l *fileStore) load() error {
 	return nil
 }
 
-func (l *fileStore) Set(services map[string][]string) {
-	panic("TO BE IMPLEMENTED")
+func (l *fileStore) Set(services map[string][]string) error {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	// no error will return from memorystore.set
+	_ = l.memoryStore.Set(services)
+
+	f, err := os.OpenFile(l.persistentFile, os.O_TRUNC|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+
+	s := ""
+	for name, addrs := range services {
+		for _, addr := range addrs {
+			s += fmt.Sprintf("%s&%s\n", name, addr)
+		}
+	}
+
+	_, err = f.WriteString(s)
+	if err != nil {
+		return err
+	}
+
+	_ = f.Sync()
```

We should not ignore the error